### PR TITLE
JAMES-2708 Simplified proposal of LinShare Docker-Compose for testing

### PR DESCRIPTION
**Pull request by chibenwa**
JAMES-2708 Simplified proposal of LinShare Docker-Compose for testing

### CODE REVIEW

**Comment by chibenwa** [comment]
We need to wait the `database-init` to be promoted as offical docker image before moving forward on this.

**Comment by trantienduchn** [comment]
:+1:

File: third-party/linshare/src/test/resources/logback-test.xml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<configuration>
```
Comment (trantienduchn) [inline third-party/linshare/src/test/resources/logback-test.xml]: apache license?

File: third-party/linshare/src/test/java/org/apache/james/linshare/LinshareTest.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                .build();
+        RestAssured.enableLoggingOfRequestAndResponseIfValidationFails();
+    }
+
+    @Test
+    void linshareShouldStart() {
+        given()
+            .auth().basic("user1@linshare.org", "password1")
+        .when()
+            .log().uri()
```
Comment (trantienduchn) [inline third-party/linshare/src/test/java/org/apache/james/linshare/LinshareTest.java]: we might not need it anymore

File: third-party/linshare/src/test/java/org/apache/james/linshare/LinshareTest.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            .get("linshare/webservice/rest/user/v2/documents")
+            .prettyPeek()
```
Comment (trantienduchn) [inline third-party/linshare/src/test/java/org/apache/james/linshare/LinshareTest.java]: idem

File: third-party/linshare/src/test/java/org/apache/james/linshare/LinshareTest.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+import io.restassured.RestAssured;
+import io.restassured.builder.RequestSpecBuilder;
+import io.restassured.http.ContentType;
+
+@ExtendWith(LinshareExtension.class)
+class LinshareTest {
+    @BeforeEach
+    void setup(DockerComposeContainer<?> linshare) {
+        RestAssured.requestSpecification = new RequestSpecBuilder()
+                .setContentType(ContentType.JSON)
```
Comment (trantienduchn) [inline third-party/linshare/src/test/java/org/apache/james/linshare/LinshareTest.java]: can you fix the indentation?

**Comment by trantienduchn** [comment]
JIRA ticket in commit messages should be `JAMES-2720`

File: third-party/linshare/src/test/java/org/apache/james/linshare/LinshareExtension.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+import org.testcontainers.containers.wait.strategy.Wait;
+
+public class LinshareExtension implements BeforeAllCallback, AfterAllCallback, ParameterResolver {
+    private static final String WAIT_FOR_LOG_MSG_PATTERN = ".*/linshare/webservice/rest/admin/authentication/change_password.*";
+    public static final String BASE_PATH = "/linshare/webservice/rest/user/v2";
+    private static final String DOCKER_COMPOSE_YML = "docker-compose.yml";
+    public static final String LINSHARE_BACKEND_SERVICE = "backend";
+    public static final int LINSHARE_BACKEND_PORT = 8080;
+
+    private DockerComposeContainer<?> linshare;
```
Comment (trantienduchn) [inline third-party/linshare/src/test/java/org/apache/james/linshare/LinshareExtension.java]: can you have a DockerLinshare class to wrap it, to hide the complex of testcontainers docker-compose usage?
We already have similar classes like `DockerCassandra` and `DockerRabbitMQ`

**Comment by hoangdat** [comment]
Read it

**Comment by chibenwa** [comment]
```
[0ecb4e6a04f01191d1571640e16b90dcf3ce895a] [INFO] -------------------------------------------------------
[0ecb4e6a04f01191d1571640e16b90dcf3ce895a] [INFO]  T E S T S
[0ecb4e6a04f01191d1571640e16b90dcf3ce895a] [INFO] -------------------------------------------------------
[0ecb4e6a04f01191d1571640e16b90dcf3ce895a] [INFO] Running org.apache.james.linshare.LinshareTest
[0ecb4e6a04f01191d1571640e16b90dcf3ce895a] [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.548 s <<< FAILURE! - in org.apache.james.linshare.LinshareTest
[0ecb4e6a04f01191d1571640e16b90dcf3ce895a] [ERROR] linshareShouldStart  Time elapsed: 0.541 s  <<< ERROR!
[0ecb4e6a04f01191d1571640e16b90dcf3ce895a] java.lang.NullPointerException
[0ecb4e6a04f01191d1571640e16b90dcf3ce895a] 	at org.apache.james.linshare.LinshareTest.setup(LinshareTest.java:46)
[0ecb4e6a04f01191d1571640e16b90dcf3ce895a] 
[0ecb4e6a04f01191d1571640e16b90dcf3ce895a] [INFO] 
[0ecb4e6a04f01191d1571640e16b90dcf3ce895a] [INFO] Results:
[0ecb4e6a04f01191d1571640e16b90dcf3ce895a] [INFO] 
[0ecb4e6a04f01191d1571640e16b90dcf3ce895a] [ERROR] Errors: 
[0ecb4e6a04f01191d1571640e16b90dcf3ce895a] [ERROR]   LinshareTest.setup:46 NullPointer
[0ecb4e6a04f01191d1571640e16b90dcf3ce895a] [INFO] 
[0ecb4e6a04f01191d1571640e16b90dcf3ce895a] [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0
```

We might have to investigate why CI can't run this docker-compose

**Comment by chibenwa** [comment]
test this please

**Comment by Arsnael** [comment]
test this please

**Comment by chibenwa** [comment]
See https://github.com/linagora/james-project/pull/2308

**Pull request closed without merging**